src/main/java/net/dv8tion/jda/api/entities/Message.java

```diff
@@ -555,17 +555,60 @@ default String getApplicationId()
     @Nonnull
     Guild getGuild();
 
+    /**
+     * Returns {@code true} if this message has any attachments.
+     *
+     * <p><b>Requires {@link net.dv8tion.jda.api.requests.GatewayIntent#MESSAGE_CONTENT GatewayIntent.MESSAGE_CONTENT}</b>
+     *
+     * @return True, if this message has attachments.
+     *
+     * @see    #getAttachments()
+     */
+    boolean hasAttachments();
+
     /**
      * An immutable list of {@link net.dv8tion.jda.api.entities.Message.Attachment Attachments} that are attached to this message.
      * <br>Most likely this will only ever be 1 {@link net.dv8tion.jda.api.entities.Message.Attachment Attachment} at most.
      *
      * <p><b>Requires {@link net.dv8tion.jda.api.requests.GatewayIntent#MESSAGE_CONTENT GatewayIntent.MESSAGE_CONTENT}</b>
      *
      * @return Immutable list of {@link net.dv8tion.jda.api.entities.Message.Attachment Attachments}.
+     *
+     * @see    #hasAttachments()
+     * @see    #getAttachmentsByExtension(String)
      */
     @Nonnull
     List<Attachment> getAttachments();
 
+    /**
+     * Returns {@code true} if this message has any attachments with the provided extension.
+     *
+     * <p><b>Requires {@link net.dv8tion.jda.api.requests.GatewayIntent#MESSAGE_CONTENT GatewayIntent.MESSAGE_CONTENT}</b>
+     *
+     * @param extension The extension to filter by (ex. "png")
+     *
+     * @return True, if this message has an attachment with the provided extension.
+     *
+     * @see    #getAttachmentsByExtension(String)
+     */
+    boolean hasAttachmentsByExtension(@Nonnull String extension);
+
+    /**
+     * An immutable list of {@link net.dv8tion.jda.api.entities.Message.Attachment Attachments} that are attached to this message, filtered by file extension.
+     * <br>Most likely this will only ever be 1 {@link net.dv8tion.jda.api.entities.Message.Attachment Attachment} at most.
```

Comment:
Why would there be at most 1 attachment with the same extension ?

Reply:
This warning was included in getAttachments() so I thought I'll just add it to this as well

Reply:
Yea I guess, but desktop users aren't limited to 1 attachment anymore, might as well remove it imo

Reply:
Sure